src/main/java/io/cryostat/rules/RuleRegistry.java

```diff
@@ -183,6 +183,17 @@ public void enableRule(Rule rule, boolean enabled) {
         String name = rule.getName();
         if (enabled != rule.isEnabled()) {
             getRule(name).get().setEnabled(enabled);
+            Path destination = rulesDir.resolve(rule.getName() + ".json");
+            try {
+                this.fs.writeString(
```

Comment:
Since this is pretty much the same thing that is done in `addRule()`, could you extract this to a helper method? Something like `private void persistRule(Rule rule)` probably, which just writes the JSON form of the rule to the corresponding path like we have here.

Reply:
Done!